src/vumps_localham.jl

```diff
@@ -19,35 +19,63 @@ struct Hᴬᶜ
   n::Int
 end
 
+
+#Assume all local Hamiltonians have the same nrange
 function Base.:*(H::Hᶜ, v::ITensor)
   ∑h = H.∑h
   Hᴸ = H.Hᴸ
   Hᴿ = H.Hᴿ
   ψ = H.ψ
   ψ′ = dag(ψ)'
+  Nsites = nsites(ψ)
+  range_∑h = nrange(∑h, 1)
   n = H.n
   l = linkinds(only, ψ.AL)
   l′ = linkinds(only, ψ′.AL)
   r = linkinds(only, ψ.AR)
   r′ = linkinds(only, ψ′.AR)
+  s = siteinds(only, ψ)
+  s′ = siteinds(only, ψ′)
+  δˢ(n) = δ(s[n], s′[n])
+
   δˡ = δ(l[n], l′[n])
-  δˡ⁻¹ = δ(l[n - 1], l′[n - 1])
   δʳ = δ(r[n], r′[n])
-  δʳ⁺¹ = δ(r[n + 1], r′[n + 1])
   Hᶜᴸv = v * Hᴸ[n] * dag(δʳ)
   Hᶜᴿv = v * δˡ * Hᴿ[n]
-  Hᶜʰv =
-    v * ψ.AL[n] * δˡ⁻¹ * ψ′.AL[n] * ∑h[(n, n + 1)] * ψ.AR[n + 1] * dag(δʳ⁺¹) * ψ′.AR[n + 1]
+  # TODO do we prefer a formalism like the commented one? if yes, need to define contraction of ITensor and MPS
+  #Hᶜʰv = v * ψ.AL[n] * δ(l[n - 1], l′[n - 1]) * ψ′.AL[n] * ∑h[(n, n + 1)] *
+  #  ψ.AR[n+1:n + range_∑h - 1]... * ψ′.AR[n+1:n + range_∑h - 1]... * dag(δ(r[n+range_∑h - 1], r′[n+range_∑h - 1]))
```

Comment:
I guess that kind of notation could be nice, but we would have to be careful about proper order of operations for efficiency. I think the code below is reasonable, and we could consider more clever notation later.